### PR TITLE
chore: reorganize the project layout and add an example of .skywalking.yaml

### PR DESCRIPTION
**Pull request by kezhenxu94**
This patch reshapes the project layout by following https://github.com/golang-standards/project-layout/ , to prepare for our NGE2E project, this may need a little changes in the `swck` @hanahmily because the schema is moved to `/api`, which I think is more suitable.

### CODE REVIEW

File: examples/skywalking.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+base-url: http://demo.skywalking.apache.org/graphql
+username: basic-auth-username
+password: basic-auth-password
```
Comment (fgksgf) [inline examples/skywalking.yaml:20]: I think this file's name should be `.skywalking.yaml`

Comment (kezhenxu94) [reply examples/skywalking.yaml]: done

File: pkg/heatmap/axes/label.go [hunk 2]
```diff
@@ -108,7 +108,7 @@ func xLabels(yEnd image.Point, graphWidth int, stringLabels []string, cellWidth
 
 // paddedLabelLength calculates the length of the padded label and
 // the column index corresponding to the label.
-// For example, the longest label's length is 5, like '12:34', and the cell's width is 3.
+// For examples, the longest label's length is 5, like '12:34', and the cell's width is 3.
```
Comment (fgksgf) [inline pkg/heatmap/axes/label.go:111]: I don't think `for examples` is correct.
image
image

Comment (kezhenxu94) [reply pkg/heatmap/axes/label.go:111]: Neither do I, the IDE changed too many

**Review by wu-sheng — DISMISSED** [review]
Make sense to me.

**Comment by hanahmily** [comment]
Noted. Actually, SWCK needs functions in `pkg` with an error return value, instead of exiting the process when facing some errors as below 

```
func AllServices(cliCtx *cli.Context, duration api.Duration) ([]api.Service, error)
```

Do you have any convenient path to support it? 
The only downside of the current solution is SWCK has to copy the implementation of these functions, which brings some issues if the details of them are updated.

**Comment by kezhenxu94** [comment]
> Noted. Actually, SWCK needs functions in `pkg` with an error return value, instead of exiting the process when facing some errors as below
> 
> ```
> func AllServices(cliCtx *cli.Context, duration api.Duration) ([]api.Service, error)
> ```
> 
> Do you have any convenient path to support it?
> The only downside of the current solution is SWCK has to copy the implementation of these functions, which brings some issues if the details of them are updated.

I see, we have this 

https://github.com/apache/skywalking-cli/blob/2ed3443179b3995f44d5588c66716fe24b6f0341/pkg/graphql/client/client.go#L39

and this

https://github.com/apache/skywalking-cli/blob/2ed3443179b3995f44d5588c66716fe24b6f0341/pkg/graphql/client/client.go#L57

the current query (`AllServices`, etc.) calls the latter one, so I think we can call the former one and let the `command` package to `panic` when there is an error, this should meet your requirement right? @hanahmily

**Comment by kezhenxu94** [comment]
> > Noted. Actually, SWCK needs functions in `pkg` with an error return value, instead of exiting the process when facing some errors as below
> > ```
> > func AllServices(cliCtx *cli.Context, duration api.Duration) ([]api.Service, error)
> > ```
> > 
> > 
> > Do you have any convenient path to support it?
> > The only downside of the current solution is SWCK has to copy the implementation of these functions, which brings some issues if the details of them are updated.
> 
> I see, we have this
> 
> https://github.com/apache/skywalking-cli/blob/2ed3443179b3995f44d5588c66716fe24b6f0341/pkg/graphql/client/client.go#L39
> 
> and this
> 
> https://github.com/apache/skywalking-cli/blob/2ed3443179b3995f44d5588c66716fe24b6f0341/pkg/graphql/client/client.go#L57
> 
> the current query (`AllServices`, etc.) calls the latter one, so I think we can call the former one and let the `command` package to `panic` when there is an error, this should meet your requirement right? @hanahmily

Does it matter if we have other return values apart from the `error`? Like returning two values `string, error`?

**Comment by hanahmily** [comment]
> the current query (AllServices, etc.) calls the latter one, so I think we can call the former one and let the command package to panic when there is an error, this should meet your requirement right

That's exactly what I expect, thanks zhenxu.

> Does it matter if we have other return values apart from the error?

 It doesn't matter to SWCK, go for it.

**Comment by kezhenxu94** [comment]
@hanahmily done

**Review by fgksgf — APPROVED** [review]

**Comment by kezhenxu94** [comment]
The comments have been addressed. I'm merging, if anything else need to be done, please let me know @hanahmily :)

**Pull request merged**